finality-aleph/src/new_network/connection_manager/session.rs

```diff
@@ -94,19 +104,29 @@ impl Handler {
         if self.peers_by_node.len() + 1 == node_count {
             return Vec::new();
         }
-        (0..node_count)
-            .map(NodeIndex)
-            .filter(|node_id| *node_id != self.index() && !self.peers_by_node.contains_key(node_id))
-            .collect()
+        match self.index() {
+            Some(index) => (0..node_count)
+                .map(NodeIndex)
+                .filter(|node_id| *node_id != index && !self.peers_by_node.contains_key(node_id))
+                .collect(),
+            _ => (0..node_count)
+                .map(NodeIndex)
+                .filter(|node_id| !self.peers_by_node.contains_key(node_id))
+                .collect(),
+        }
     }
 
     /// Verifies the authentication, uses it to update mappings, and returns whether we should
     /// remain connected to the multiaddresses.
     pub fn handle_authentication(&mut self, authentication: Authentication) -> bool {
         let (auth_data, signature) = authentication.clone();
-        if auth_data.session_id != self.session_id() {
-            return false;
+
+        if let Some(session_id) = self.session_id() {
```

Comment:
Yeah, `session_id` definitely should return something always.

Reply:
Done